### PR TITLE
[tokenizers] Updates data processors, docstring, examples and model cards to the new API

### PR DESCRIPTION
**Pull request by thomwolf**
Updates the data-processors to the new recommended tokenizers' API instead of the old one.
Also update the docstrings, examples, and model-cards which were using the old API.

Supersede #5310

@sshleifer you have a couple of methods only your models use (bart and marian). I'm not sure about the consequences of updating those API so I'll let you update them. Here is the doc on the new tokenizer API if you need it: https://huggingface.co/transformers/master/preprocessing.html
Recommended updates:
- use `__call__` instead of `encode_plus` and `batch_encode_plus`
- use `padding` and `truncation` instead of `max_length` only and `pad_to_max_length`

### CODE REVIEW

**Comment by thomwolf** [comment]
@sshleifer @patrickvonplaten and @yjernite I updated your examples (seq2seq and eli5). Maybe you want to check it.

**Comment by yjernite** [comment]
> @sshleifer @patrickvonplaten and @yjernite I updated your examples (seq2seq and eli5). Maybe you want to check it.

lgtm!

File: examples/seq2seq/utils.py [hunk 1]
```diff
@@ -41,12 +41,12 @@ def encode_file(
     assert lns, f"found empty file at {data_path}"
     examples = []
     for text in tqdm(lns, desc=f"Tokenizing {data_path.name}"):
-        tokenized = tokenizer.batch_encode_plus(
+        tokenized = tokenizer(
```
Comment (sshleifer) [inline examples/seq2seq/utils.py:44]: LGTM, thanks!

**Review by sshleifer — APPROVED** [review]
